input_buffer.cc

```diff
@@ -874,6 +882,19 @@ expression_ptr text_input_buffer::parse_expression(bool stopAtParen)
 	source_location l = location();
 	switch (*(*this))
 	{
+		case '\'':
```

Comment:
I think that most of this logic would make more sense in `consume_char_literal`, but I don't have a very strong opinion here.